### ISSUE TITLE
Invalid pool key

### ISSUE THREAD

**Issue by edamame43** [madMAx43v3r/chia-plotter#78]
Chia plot command showing “ Invalid <pool_key>: 00 (needs to be 48 bytes)” error. 
I triple checked the keys are not wrong. What’s going on here?

**Comment by grayfallstown** [comment]
Please post the command line you use to call chia_plot

**Comment by madMAx43v3r** [comment]
I had that once on a particular commit, are you on latest master?

**Comment by hhdllhflower** [comment]
Invalid <farmerkey>: 00000e0000000d00010000 (needs to be 48 bytes)

**Comment by edamame43** [comment]
Yes latest version. Although I used a pre compiled executable, might be the reason?

**Comment by madMAx43v3r** [comment]
could be, could also be a mistake on the command line, can you post the line?

**Comment by grayfallstown** [comment]
> Yes latest version. Although I used a pre compiled executable, might be the reason?

Precompiled? Are you using the windows binaries? They don't support -p and so on, the windows version is still using the old positional parameters: https://github.com/stotiks/chia-plotter/tree/v0.0.1-alpha#usage

**Comment by edamame43** [comment]
Don’t know if it’s compiled on win. Will try to compile again later

**Comment by hhdllhflower** [comment]
@madMAx43v3r ./chia_plot -p 860a6c3074435b1efe6ed745cbd9440e8339940b6bf10f1140d52ac71736cc042027569499b87950d525ef7e29269a52 -f 860a6c3074435b1efe6ed745cbd9440e8339940b6bf10f1140d52ac71736cc042027569499b87950d525ef7e29269a52 -n 1 -r 16 -t /home/sandy/disk1/tmp -f /home/sandy/disk1/new

Invalid <farmerkey>: 00000e0000000d00010000 (needs to be 48 bytes)

**Comment by madMAx43v3r** [comment]
`-f /home/sandy/disk1/new`

=> that should be `-d`

**Comment by edamame43** [comment]
For me it’s Invalid : 00 (needs to be 48 bytes)

**Comment by hhdllhflower** [comment]
> `-f /home/sandy/disk1/new`
> 
> => that should be `-d`

thanks

**Comment by madMAx43v3r** [comment]
@edamame43 can you post the line?

**Comment by madMAx43v3r** [comment]
@hhdllhflower also paths need a trailing slash

**Comment by viponedream** [comment]
chia_plot -p 80d769219b03f1f392327340c1a2b879d46852932764aa1d20d08110422908fa6a874d7fe9baf4ea668c18a852c15905 -f b7fb3f03444796ee4689a188b7beba727a34e831dcde182fa273f014870c08646687456aee7efde5e637c81273e15a3c  -t G:/tmp -d F:/FARM-2

Invalid <pool_key>: 00 (needs to be 48 bytes)

**Comment by madMAx43v3r** [comment]
```
./build/chia_plot -p 80d769219b03f1f392327340c1a2b879d46852932764aa1d20d08110422908fa6a874d7fe9baf4ea668c18a852c15905 -f b7fb3f03444796ee4689a188b7beba727a34e831dcde182fa273f014870c08646687456aee7efde5e637c81273e15a3c -t G:/tmp/ -d F:/FARM-2/
Failed to write to <tmpdir> directory: 'G:/tmp/'
```
Works on my machine... of course I don't have those drives.

**Comment by edamame43** [comment]
Didn’t save it. Was testing from my pen drive. Could be the trailing slash... it’s being so specific about pool key I might overlook syntax

**Comment by vasya37** [comment]
![image](https://user-images.githubusercontent.com/46578285/121398007-94c4d080-c95d-11eb-9ca5-ec297ae521eb.png)

**Comment by vasya37** [comment]
![image](https://user-images.githubusercontent.com/46578285/121398185-c2117e80-c95d-11eb-9c23-64f52aeba726.png)
I have the second option starts working and immediately stops.

**Comment by madMAx43v3r** [comment]
must be some windows issue... i've add some more output to debug in 99783dfe23ab5f50ef7318a28d348f0da27c562a

**Comment by madMAx43v3r** [comment]
ah you have the old command line, no idea why it just exits right away..

**Comment by vasya37** [comment]
How do I edit the. exe file?

**Comment by vasya37** [comment]
I downloaded the exe from here and run it, am I doing something wrong? 
https://github.com/stotiks/chia-plotter/releases/tag/v0.0.1-alpha

**Comment by grayfallstown** [comment]
> https://github.com/stotiks/chia-plotter/tree/v0.0.1-alpha#usage
> Had the same error with them.

The windows version is still using the old positional parameters. Do not use -p, -t, -2, etc, just pass the values by position.

**Comment by vasya37** [comment]
![image](https://user-images.githubusercontent.com/46578285/121399255-ec177080-c95e-11eb-8d90-5e0c210f99d8.png)

**Comment by vasya37** [comment]
Strangely, it stops immediately.

**Comment by armycats** [comment]
Dude, why are you asking the original author for something somebody else compiled for Windows? :D

Works in PowerShell, by the way.

**Comment by vasya37** [comment]
At some point I thought that the author did it for windows.

**Comment by geoffwalmsley** [comment]
sorry to jump on this issue - but I am getting the invalid pool key error even when I don't specify it

./chia_plot -n 1 -r 32 -t /mnt/ssd/ -d /mnt/hdd/

Invalid <pool_key>: 00 (needs to be 48 bytes)

**Comment by stotiks** [comment]
specify it

**Comment by geoffwalmsley** [comment]
> specify it

haha, sorry - I should have said I also get the error when specifying it

**Comment by canpo41** [comment]
I received below erro please help ?
Set-Location : A positional parameter cannot be found that accepts argument 'poyraz\Desktop\madMAx'.
At line:1 char:1
+ cd C:\Users\can poyraz\Desktop\madMAx
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-Location], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.SetLocationCommand

**Comment by VertiHydro** [comment]
> I received below erro please help ?
> Set-Location : A positional parameter cannot be found that accepts argument 'poyraz\Desktop\madMAx'.
> At line:1 char:1
> 
> * cd C:\Users\can poyraz\Desktop\madMAx
> * ```
>     + CategoryInfo          : InvalidArgument: (:) [Set-Location], ParameterBindingException
>     + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.SetLocationCommand
>   ```

Firstly, this is a Windows problem, not a Madmax problem, but the solution is that you need to put the path in quotes because there is a space between "can" and "poyraz".
Therefore, the command would be:
`cd "C:\Users\can poyraz\Desktop\madMAx"`
